.travis/install.sh

```diff
@@ -62,26 +62,22 @@ else
         pyenv global pypy-4.0.1
     fi
     if [[ "${OPENSSL}" == "0.9.8" ]]; then
-      # download, compile, and install if it's not already present via travis cache
-      # This is 0.9.8l rather than zh because we have some branches for handling
-      # < 0.9.8m that won't be exercised with a newer OpenSSL. (RHEL5 is 0.9.8e with
-      # patches, but while that's in jenkins we don't get coverage data from it)
-      if [[ ! -f "$HOME/ossl-098l/bin/openssl" ]]; then
-        curl -O https://www.openssl.org/source/openssl-0.9.8l.tar.gz
-        tar zxf openssl-0.9.8l.tar.gz
-        cd openssl-0.9.8l
-        echo "OPENSSL_0.9.8L_CUSTOM {
-            global:
-              *;
-        };" > openssl.ld
-        ./config no-asm no-ssl2 -Wl,--version-script=openssl.ld -Wl,-Bsymbolic-functions -fPIC shared --prefix=$HOME/ossl-098l
-        make depend
-        make install
-      fi
-      export PATH="$HOME/ossl-098l/bin:$PATH"
-      export CFLAGS="-I$HOME/ossl-098l/include"
-      export LDFLAGS="-L$HOME/ossl-098l/lib"
-      export LD_LIBRARY_PATH="$HOME/ossl-098l/lib"
+        # We use 0.9.8l rather than zh because we have some branches for handling
+        # < 0.9.8m that won't be exercised with a newer OpenSSL. (RHEL5 is 0.9.8e with
+        # patches, but while that's in jenkins we don't get coverage data from it).
+        OPENSSL_VERSION_NUMBER="0.9.8l"
+        OPENSSL_DIR="ossl-098l"
+    fi
+    # download, compile, and install if it's not already present via travis cache
+    if [ ! -z "$OPENSSL_DIR" ]; then
+        if [[ ! -f "$HOME/$OPENSSL_DIR/bin/openssl" ]]; then
```

Comment:
If we don't check to see if `OPENSSL_DIR` is set then it will check for `$HOME//bin/openssl` which will always return false. It will then try to download openssl, but `OPENSSL_VERSION_NUMBER` won't be set either so this will all fail.

The reason to separate them is so when we add 1.0.0 we can just add an `elif [[ "${OPENSSL}" == "1.0.0" ]]; then` to the top conditional and then share all the logic past that.